### ISSUE TITLE
TODO list

### ISSUE THREAD

**Issue by nikkolasg** [Lagrange-Labs/deep-prove#17]
- [ ] Requantization 
    - [ ] formula (list of scaling factors depending on model) (shravan WIP)
    - [ ] Integration in RELU - Each relu step should take the formula as input when describing the layer
- [ ] Lookup Integration
    - [ ] lookup implementation (zac WIP)
    - [ ] accumulation of claims from lookups (nikko WIP)
    - [ ] links between dense and lookups (nikko WIP)
- [ ] Move everything to basefield - only claims on extension field
- [ ] Benchmarking 
    - [ ] Need a _large_ model in onnx format: dense -> relu -> dense -> relu
    - [ ] Writing bench: setup, inference, proving and verifying
    - [ ] Testing accuracy + performance
   

Accuracy Measurement:
* Run model M quantized over any framework (pytorch, keras etc)       -> get output in O1 =  [-128;127]
* Run model M with quantized inputs and weights (on ceno)                 -> get output in O2 = [-128;127] (example)
   * we already give model to ceno in quantized format
   * we have the quantization factors after any steps ? (that "match" any framework)
* Compare the accuracy of the outputs:
   * argmax(O1) == argmax(O2) etc
   * how many position they differ, X => acc = X / N  (size of output)
